### PR TITLE
Adding ArtificialRoot to non-deferred apps for diff table

### PR DESCRIPTION
**Pull request by carolynqu**
When using the DiffTable to compare a non-deferred and deferred app, the main app roots are not at the same location. 

In order to get an accurate comparison between the two apps, the non-deferred app will add its own ArtificialRoot that deferred apps have.

### CODE REVIEW

**Comment by elliette** [comment]
Let's change the check on line 588 from:

```
    if (totalByteSize == 0) {
      return null;
    }
```
 
to 
```
    if (totalByteSize == 0 && !isDeferredApp.value) {
      return null;
    } 
```

That way if the totalByteSize stays the same (for example, part of the main unit is moved to deferred units) we still build the diff view.

File: packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart
Comment (elliette) [inline packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart:70]: Let's rename this to `artificialRootNodeName`. Let's also add another constant `mainNodeName` which is equal to `Main`

File: packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart
Comment (elliette) [inline packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart:392]: Let's pull this logic out into a private method (eg, `_wrapInArtificialRoot`) which takes the json file as a parameter. That way we don't have to repeat the code here and below. 

I think we also have to rename the first node from `Root` to `Main` before we wrap it, ie:

```
          oldFileJson['n'] = 'Main';
          oldFileJson = <String, dynamic>{
            'n': deferredRootName,
            'children': [oldFileJson]
          };
```

**Comment by elliette** [comment]
> Let's change the check on line 588 from:
> 
> ```
>     if (totalByteSize == 0) {
>       return null;
>     }
> ```
> 
> to
> 
> ```
>     if (totalByteSize == 0 && !isDeferredApp.value) {
>       return null;
>     } 
> ```
> 
> That way if the totalByteSize stays the same (for example, part of the main unit is moved to deferred units) we still build the diff view.

Thinking about this more, we only want to show 0 byte size changes on the root node (not on the child nodes). Instead, let's add another parameter to both `generateDiffTree` and `_buildNodeWithChildren` (eg `skipNodesWithNoByteSizeChange`) which defaults to `true`. When we [generate the diff trees here](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart#L444-L455)`skipNodesWithNoByteSizeChange` should be false if deferred app is true. For every recursive call on child nodes, `skipNodesWithNoByteSizeChange` should be the default value of true.

File: packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart
Comment (elliette) [inline packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart:20]: Switch to false :)

**Review by elliette — APPROVED** [review]
LGTM once the trailing comma the linter is complaining about is added

File: packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart:72]: we should use the functionality @elliette added in https://github.com/flutter/devtools/pull/4346 once that lands.

**Pull request merged**
